doc/collision_detection.md

```diff
@@ -0,0 +1,161 @@
+can use the `Hitbox` mixin to define a more accurate polygon (or another shape) for which the input
+should be within for the event to be registered on your component.
+
+You can add new shapes to the `Hitbox` just like they are added in the below `Collidable` example.
+
+### Collidable
+The `Collidable` mixin is added to a `PositionComponent` that has a `HitBox` and it is used for
+detecting collisions with other `Collidable`s. To make your component collidable you would start off
+like this:
+
+```dart
+class MyCollidable extends PositionComponent with Hitbox, Collidable {
+  MyCollidable (
+    // This could also be done in onLoad instead of in the constructor
+    final shape = HitboxPolygon([
+      Vector2(0, 1),
+      Vector2(1, 0),
+      Vector2(0, -1),
+      Vector2(-1, 0),
+    ]);
+    addShape(shape);
+  }
+}
+```
+
+The `HitboxPolygon` added to the `Collidable` here is a diamond shape(◇).
+More about how the different shapes are defined in the [Shapes](#Shapes) section.
+
+Remember that you can add as many `HitboxShape`s as you want to your `Collidable` to make up more
+complex hitboxes. For example a snowman with a hat could be represented by three `HitboxCircle`s and
+a `HitboxPolygon` as its hat.
+
+To react to a collision you should override the `collisionCallback` in your component.
+Example:
+
+```dart
+class MyCollidable extends PositionComponent with Hitbox, Collidable {
+  ...
+
+  @override
+  void collisionCallback(Set<Vector2> points, Collidable other) {
+    if (other is CollidableScreen) {
+      ...
+    } else if (other is YourOtherCollidable) {
+      ...
+    }
+  }
+}
+```
+
+In this example it can be seen how the Dart `is` keyword is used to check which other `Collidable`
+that your component collided with. The set of points is where the hitboxes of the collidables
+intersected. If you want to check collisions with the screen edges you can use the predefined
+[ScreenCollidable](#ScreenCollidable) class.
+
+### HasCollidables
+If you want to use this collision detection in your game you have to add the `HasCollidables` mixin
+to your game so that the game knows that it should keep track of which components that can collide.
+
+Example:
+```dart
+class MyGame extends BaseGame with HasCollidables {
+  ...
+}
+```
+
+Now when you add your `Collidable` components to your game they will automatically be checked for
+collisions.
+
+### ScreenCollidable
+`ScreenCollidable` is not a mixin, but a pre-made collidable component which represents the edges of
+your game/screen. If you add a `ScreenCollidable` to your game your other collidables will be
+notified when they collide with the edges. It doesn't take any arguments, it only depends on the
+`size` of the game that it is added to. To add it you can just do `add(ScreenCollidable())` in
+your game if you don't want the `ScreenCollidable` itself to be notified when something collides
```

Comment:
I think that this section `if you don't want ...` should have a comma or a full stop.